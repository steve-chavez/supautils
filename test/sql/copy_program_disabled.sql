-- create a superuser role
create role super2 superuser;
set role super2;
\echo

-- superuser can't execute COPY ... PROGRAM
copy (select '') to program 'id';
\echo

-- superuser can't disable guc
set supautils.disable_program = false;
\echo

-- current_setting must be 'on'
select current_setting('supautils.disable_program');
alter system set supautils.disable_program = false;
select  pg_reload_conf();
--  altering the system config does  not change the state of supautils.disable_program
select current_setting('supautils.disable_program');
\echo

-- cleanup
set role postgres;
drop role super2;
\echo
