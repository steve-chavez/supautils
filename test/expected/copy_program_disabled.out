-- create a superuser role
create role super2 superuser;
set role super2;
\echo

-- superuser can't execute COPY ... PROGRAM
copy (select '') to program 'id';
ERROR:  COPY TO/FROM PROGRAM not allowed
DETAIL:  The copy to/from program utility statement is disabled
\echo

-- superuser can't disable guc
set supautils.disable_program = false;
ERROR:  parameter "supautils.disable_program" cannot be changed now
\echo

-- current_setting must be 'on'
select current_setting('supautils.disable_program');
 current_setting 
-----------------
 on
(1 row)

alter system set supautils.disable_program = false;
select  pg_reload_conf();
 pg_reload_conf 
----------------
 t
(1 row)

--  altering the system config does  not change the state of supautils.disable_program
select current_setting('supautils.disable_program');
 current_setting 
-----------------
 on
(1 row)

\echo

-- cleanup
set role postgres;
drop role super2;
\echo

